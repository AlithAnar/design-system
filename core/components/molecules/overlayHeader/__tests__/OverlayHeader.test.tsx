import * as React from 'react';
import { render } from '@testing-library/react';
import OverlayHeader, { OverlayHeaderProps as Props } from '../OverlayHeader';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const Heading = 'Modal header';
const SubHeading = 'Modal subheading';
const onClose = jest.fn();

const Mapper = {
  heading: valueHelper(Heading, { required: true }),
  onClose: valueHelper(onClose, { required: true }),
  subHeading: valueHelper(SubHeading, { required: true }),
};

describe('OverlayHeader component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { asFragment } = render(
        <OverlayHeader
          {...attr}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  };

  testHelper(Mapper, testFunc);
});

describe('OverlayHeader component with props', () => {

  it('renders subheading', () => {
    const { getByTestId } = render(<OverlayHeader heading={Heading} onClose={onClose} subHeading={SubHeading} />);
    expect(getByTestId('DesignSystem-OverlayHeader--Subheading')).toBeInTheDocument();
  });

});

describe('OverlayHeader with overwrite class', () => {
  const className = 'DS-OverlayHeader';

  it('overwrite OverlayHeader class', () => {
    const { getByTestId } = render(<OverlayHeader heading={Heading} onClose={onClose} className={className} />);
    expect(getByTestId('DesignSystem-OverlayHeader')).toHaveClass(`OverlayHeader ${className}`);
  });

});
