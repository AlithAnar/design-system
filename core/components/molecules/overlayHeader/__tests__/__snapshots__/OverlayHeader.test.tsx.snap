// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayHeader component 
  heading: "Modal header", onClose: "[Function]", subHeading: "Modal subheading"
 1`] = `
<DocumentFragment>
  <div
    class="OverlayHeader"
    data-test="DesignSystem-OverlayHeader"
  >
    <div
      class="OverlayHeader-headerWrapper"
    >
      <h4
        class="Heading Heading--m Heading--default"
        data-test="DesignSystem-Heading"
      >
        Modal header
      </h4>
    </div>
    <span
      class="Text Text--subtle Text--regular OverlayHeader-subheading"
      data-test="DesignSystem-OverlayHeader--Subheading"
    >
      Modal subheading
    </span>
  </div>
</DocumentFragment>
`;
